drivers/serial/Kconfig.pl011

```diff
@@ -20,4 +20,13 @@ config UART_PL011_PORT1
 	help
 	  Build the driver to utilize UART controller Port 1.
 
+config UART_PL011_NO_CONFIG
+	bool "Do not configure PL011 registers"
+	default n
+	help
+	  Do not write anything to PL011 control registers. This setting is
+	  needed for cases when PL011 is already configured by a bootloader of
+	  for cases when PL011 is emulated by hypervisor and configuring PL011
+	  makes no sense.
```

Comment:
I don't really understand why a PL011 emulated by an hypervisor implies there are no need for configuration.

For instance, there are no promise that UART registers (such as IMCS, ICR) will be initializing a value expected by Zephyr. Imagine there was a firmware running before hand which configured the UART differently.

Reply:
Well, current vpl011 implementation in Xen lacks emulation of CR and LR_H registers.  Actually, I have a patch for Xen that handles reads and writes to those registers. But ignores what was written, so this is not a correct emulation.

On other hand, you are rigth and I should not omit configuration of IMCS and ICR from Zephyr side.

Reply:
We should try to fix this in xen instead, i agree with Julien here.
That would make it easier for other RTOS to run on top of Xen "unmodified"

Reply:
Bertrand, I agree with you that it would be great to have full PL011 emulation in Xen. But correct emulation requires a lot of efforts and I have no resources to implement it right now.

Instead, I added "SBSA mode"  to existing PL011 driver in Zephyr. I tried to align it with existing code, so I hope, this will be enough.